commit bf0e97aa0e9c7658c774d4be1c8ff3cbe2698eee
Author: Howard Chu <hyc@symas.com>
Commit: Howard Chu <hyc@symas.com>

    ITS#7718 invalidate cursor when DB gets emptied

diff --git a/libraries/liblmdb/mdb.c b/libraries/liblmdb/mdb.c
index ddde69d..e935dc6 100644
--- a/libraries/liblmdb/mdb.c
+++ b/libraries/liblmdb/mdb.c
@@ -7119,6 +7119,7 @@ mdb_rebalance(MDB_cursor *mc)
 			/* Adjust cursors pointing to mp */
 			mc->mc_snum = 0;
 			mc->mc_top = 0;
+			mc->mc_flags &= ~C_INITIALIZED;
 			{
 				MDB_cursor *m2, *m3;
 				MDB_dbi dbi = mc->mc_dbi;
@@ -7132,6 +7133,7 @@ mdb_rebalance(MDB_cursor *mc)
 					if (m3->mc_pg[0] == mp) {
 						m3->mc_snum = 0;
 						m3->mc_top = 0;
+						m3->mc_flags &= ~C_INITIALIZED;
 					}
 				}
 			}
